common/src/main/java/com/wynntils/models/label/type/LabelInfo.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.label.type;
+
+import com.google.common.collect.ComparisonChain;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.utils.mc.type.Location;
+
+public abstract class LabelInfo implements Comparable<LabelInfo> {
+    protected final transient StyledText label;
+
+    protected final String name;
+    protected final String formattedLabelString;
+    protected final Location location;
+
+    protected LabelInfo(StyledText label, Location location) {
+        this.label = label;
+        this.name = label.getStringWithoutFormatting();
+        this.formattedLabelString = label.getString();
```

Comment:
Do you really need formattedLabelString? What does it provide that label does not?

Reply:
More of an "in case" if we ever need it during post processing

Reply:
Can't you just call getString on the returned label then?

Reply:
That way GSON wouldn't serialize it.

Reply:
Maybe we need a json type adapter for StyledText? That could come in handy elsewhere, too...

Reply:
Hm, I could add it, but separately. This is probably needed tomorrow.

Reply:
Gonna add this now.